### PR TITLE
Create switch.xiaomi_vacuum

### PR DESCRIPTION
**Pull request by jcastro**
First revision

**Description:**

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

### CODE REVIEW

File: source/_components/switch.xiaomi_vacuum [hunk 1]
```diff
@@ -0,0 +1,47 @@
+The `xiaomi_vacuum`switch platforms allows you to control the state of your [Xiaomi Mi Robot Vacuum](http://www.mi.com/roomrobot/). 
+Current supported features are start and stop (go to dock).
+
+{% linkable_title Installation %}
```
Comment (rytilahti) [inline source/_components/switch.xiaomi_vacuum]: The installation should be unnecessary as homeassistant downloads the dependencies.

File: source/_components/switch.xiaomi_vacuum [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+mirobo discover
+```
+
+which should return something similar to this:
+
+```
+INFO:mirobo.vacuum:  IP 192.168.8.1: Xiaomi Mi Robot Vacuum - token: b'ffffffffffffffffffffffffffffffff'
+```
+
+If the value is as shown above, the vacuum has already been connected and it needs a reset.
```
Comment (rytilahti) [inline source/_components/switch.xiaomi_vacuum]: If the value is full of ffs?

File: source/_components/switch.xiaomi_vacuum [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+```
+- platform: xiaomi_vacuum
+  name: 'name of the robot'
+  host: 192.168.1.2
+  token: your-token-here
+```
+
+Configuration variables:
+- **name** (*Required*): The name of your robot
```
Comment (rytilahti) [inline source/_components/switch.xiaomi_vacuum]: Name is not required at the moment, should it be?

File: source/_components/switch.xiaomi_vacuum [hunk 4]
```diff
@@ -0,0 +1,47 @@
+The `xiaomi_vacuum`switch platforms allows you to control the state of your [Xiaomi Mi Robot Vacuum](http://www.mi.com/roomrobot/).
```
Comment (rytilahti) [inline source/_components/switch.xiaomi_vacuum]: Platform not platforms.

File: source/_components/switch.xiaomi_vacuum (hunk 3 above)
Comment (jcastro) [inline source/_components/switch.xiaomi_vacuum]: wasn't sure about the required ones, will change it to optional then!

File: source/_components/switch.xiaomi_vacuum (hunk 2 above)
Comment (jcastro) [inline source/_components/switch.xiaomi_vacuum]: yes, at least is what I experienced when running the commands. Is it not clear enough, maybe?

File: source/_components/switch.xiaomi_vacuum [hunk 5]
```diff
@@ -0,0 +1,36 @@
+The `xiaomi_vacuum`switch platform allows you to control the state of your [Xiaomi Mi Robot Vacuum](http://www.mi.com/roomrobot/). 
+Current supported features are start and stop (go to dock).
+
+{% linkable_title Getting started %}
+
+To be able to communicate with the vacuum one needs to have its IP address as well as an encryption token.
+This token is only attainable before the device has been connected over the app to your local wifi.
+
+In order to fetch the token, reset the robot, connect to its the network its announcing (rockrobo-XXXX)
+and run the following command:
```
Comment (rytilahti) [inline source/_components/switch.xiaomi_vacuum]: I'm wondering if this part is too vague (for users who do not know how to access the virtualenv/environment of homeassistant), but can't directly think of how to write it out in a nicer manner.

File: source/_components/switch.xiaomi_vacuum [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+mirobo discover
+```
+
+which should return something similar to this:
+
+```
+INFO:mirobo.vacuum:  IP 192.168.8.1: Xiaomi Mi Robot Vacuum - token: b'ffffffffffffffffffffffffffffffff'
+```
+
+If the value is as shown above, full of ffs, the vacuum has already been connected and it needs a reset.
```
Comment (rytilahti) [inline source/_components/switch.xiaomi_vacuum]: "full of ffs" sounds weird. Maybe it'd be better to rewrite it in a more user-friendly manner?

**Comment by rytilahti** [comment]
One more note, all documentation needs an information table, see https://github.com/home-assistant/home-assistant.github.io/blob/current/source/_components/neato.markdown

**Comment by jcastro** [comment]
@rytilahti oh great, thanks for letting me know! will add it in a bit!

**Comment by jcastro** [comment]
@fabaff would you mind pointing me on the right direction to know where I should upload and what sizes and formats should the component image be like?

**Comment by fabaff** [comment]
Images for brands are located in `source/images/supported_brands`. PNG is preferred because it allows a transparent background. Size doesn't really matter as the images will be scaled anyway. Something between 200px and 400px usually works well.

**Comment by jcastro** [comment]
@fabaff thanks! I tried to upload a new image but it says I have no permissions. I'm attaching it here in case you can do it?

xiaomi_vacuum

**Comment by Landrash** [comment]
File name is wrong for this documentation.

**Comment by jcastro** [comment]
@Landrash thanks for letting me know! I checked other switches and I followed same naming conventions. Would you mind telling me what's wrong in the file name?

File: source/_components/switch.xiaomi_vacuum.markdown [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+```bash 
+$ sudo su -s /bin/bash homeassistant
+$ source /srv/homeassistant/bin/activate
+```
+
+In order to fetch the token, reset the robot, connect to its the network its announcing (rockrobo-XXXX)
+and run the following command:
+
+```
+mirobo discover
```
Comment (Landrash) [inline source/_components/switch.xiaomi_vacuum.markdown]: There seems to be some part missing on how the end users obtains this software.

**Comment by Landrash** [comment]
@jcastro Filename of the documentation needed to end with .markdown. Changed it for you.

File: source/_components/switch.xiaomi_vacuum.markdown [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+```bash 
+$ sudo su -s /bin/bash homeassistant
+$ source /srv/homeassistant/bin/activate
+```
+
+In order to fetch the token, reset the robot, connect to its the network its announcing (rockrobo-XXXX)
+and run the following command:
+
+```bash
+mirobo discover
```
Comment (Landrash) [inline source/_components/switch.xiaomi_vacuum.markdown]: No information is provided on what needs to be installed for this command to be available.

**Comment by balloob** [comment]
Please address the comments by Landrash.

File: source/_components/switch.xiaomi_vacuum.markdown [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+ha_release: 0.47
+---
+
+The `xiaomi_vacuum`switch platform allows you to control the state of your [Xiaomi Mi Robot Vacuum](http://www.mi.com/roomrobot/). 
+Current supported features are start and stop (goes to dock).
+
+{% linkable_title Getting started %}
+
+To be able to communicate with the vacuum one needs to have its IP address as well as an encryption token.
+This token is only attainable before the device has been connected over the app to your local wifi.
```
Comment (trisk) [inline source/_components/switch.xiaomi_vacuum.markdown]: This sentence would be clearer as:
`You must retrieve the token from the device before using the mobile app to connect it to your wireless network.`

File: source/_components/switch.xiaomi_vacuum.markdown (hunk 8 above)
Comment (trisk) [inline source/_components/switch.xiaomi_vacuum.markdown]: `In order to fetch the token, reset the vacuum following the instructions in the manual. Then,
connect to its built-in wireless network (**rockrobo-XXXX**) with a computer that has the
[python-mirobo](https://github.com/rytilahti/python-mirobo) package installed.
Run the following command:`

File: source/_components/switch.xiaomi_vacuum.markdown [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+mirobo discover
+```
+
+which should return something similar to this:
+
+```
+INFO:mirobo.vacuum:  IP 192.168.8.1: Xiaomi Mi Robot Vacuum - token: b'ffffffffffffffffffffffffffffffff'
+```
+
+If the value is exactly as shown above, meaning a lot of f letters, then the vacuum has already been connected and it needs a reset.
```
Comment (trisk) [inline source/_components/switch.xiaomi_vacuum.markdown]:
```
The token is the value between the single quotes. If the value is exactly as shown above
(`ffffffffffffffffffffffffffffffff`), the vacuum was previously configured and it needs to be
reset to obtain a valid token.
```

**Comment by bramkragten** [comment]
@jcastro Could you also take this PR in the documentation?

**Comment by jcastro** [comment]
@bramkragten sure, will add it!

**Comment by jcastro** [comment]
@bramkragten umm I'm a bit lost on how to structure it, mostly because I'm not too much into entities, attributes, services, etc... do you think a table like the ones here would be ok? https://home-assistant.io/components/media_player.plex/

**Comment by bramkragten** [comment]
@jcastro yeah, that would work fine!

**Comment by Landrash** [comment]
@jcastro Want me to merge this as as and you'll create a new one for the services?

**Comment by jcastro** [comment]
@Landrash sure, sounds good!

**Comment by rytilahti** [comment]
Thank you @jcastro for doing the leg work and thanks @Landrash for comments and for the merge!

**Pull request merged**